packages/react/menu/src/Menu.tsx

```diff
@@ -661,15 +665,21 @@ const MenuItem = React.forwardRef((props, forwardedRef) => {
       {...itemProps}
       ref={composedRefs}
       disabled={disabled}
-      // we handle selection on `pointerUp` rather than `click` to match native menus implementation
-      onPointerUp={composeEventHandlers(props.onPointerUp, handleSelect)}
+      onClick={composeEventHandlers(props.onClick, handleSelect)}
+      onPointerDown={() => (isPointerDownRef.current = true)}
+      onPointerUp={composeEventHandlers(props.onPointerUp, (event) => {
+        // Pointer down can move to a different menu item which should activate it on pointer up.
+        // We dispatch a click for selection to allow composition with click based triggers.
+        if (!isPointerDownRef.current) event.currentTarget?.click();
+      })}
       onKeyDown={composeEventHandlers(props.onKeyDown, (event) => {
         const isTypingAhead = contentContext.searchRef.current !== '';
         if (disabled || (isTypingAhead && event.key === ' ')) return;
         if (SELECTION_KEYS.includes(event.key)) {
-          // prevent page scroll if using the space key to select an item
-          if (event.key === ' ') event.preventDefault();
-          handleSelect();
+          event.currentTarget.click();
+          // prevent default browser behaviour for selection keys, they should trigger a
+          // selection only. for example, stops page from scrolling when hitting space key.
+          event.preventDefault();
```

Comment:
This is needed for `Enter` keypress also otherwise the `Dialog` will not open. I am really stumped why this is needed though. There isn't a `keydown` event in the `Dialog.Trigger` so we're not preventing any of our implementation, just browser stuff but I dno what that "stuff" is 🙈

Any ideas?

Reply:
Not knowing the "stuff" is bugging me too. I realised that the `Dialog` does open, it's just that the close button is triggered as well, causing it to close immediately.

Could this be related to any of the implicit form submission behaviour associated with `Enter`? I can see why it would given we're not in a form context, I'm just struggling to see what else the browser could be doing.

Reply:
> it's just that the close button is triggered as well, causing it to close immediately.

This should only be happening if you hold the `Enter` key down which is how the browser works unfortunately https://codepen.io/jjenzz/pen/rNmPwqj

Is it happening when you just press enter and don't hold it?

Reply:
Let's have a quick call because I'm not seeing that here 🤔

Reply:
Oh gosh, I see what you mean now hahaha... when I *don't* put the `event.preventDefault()` in.

> without preventing default

Missed that part. Interesting... yeah okay, let me try something quick.

Reply:
Looks like the same thing I was running into here. 🤔 

https://discord.com/channels/752614004387610674/872115819490996244/872416709607309353

Reply:
> Looks like the same thing I was running into here.

Yep! we were actually using one of those examples to help debug. I was quite surprised to see the events working this way after calling `focus` but it explains what we were seeing in that thread.